### ISSUE TITLE
missing actual "Image" error

### ISSUE THREAD

**Issue by inferrna** [mosteo/optional#1]
Got
`missing actual "Image" error`
while trying to use your crate. 
My steps was
1. `alr init`
2. `alr with optional`
3. Follow your optional-demo.adb:
```Ada
with Optional.Values;

procedure MyDemo is
    package Opt_Strings is new Optional.Values (String);
```

Since I'm very new to Ada (this is my "hello.." tries) there is still a chance I did something wrong. 
However, I fixed this with the patch below, but now it requires `-gnat2022` switch to compile. 

```diff
diff --git a/src/optional-values.ads b/src/optional-values.ads
index 0789355..d78d15d 100644
--- a/src/optional-values.ads
+++ b/src/optional-values.ads
@@ -3,7 +3,6 @@ private with Ada.Finalization;
 
 generic
    type Element_Type (<>) is private;
-   with function Image (This : Element_Type) return String;
 package Optional.Values with Preelaborate is
 
    --  Reference types
@@ -99,6 +98,8 @@ package Optional.Values with Preelaborate is
    -- References --
    ----------------
 
+   function Image (This : Element_Type) return String;
+
    function Image (This : Const_Ref) return String;
 
    function Image (This : Var_Ref) return String;
@@ -182,6 +183,9 @@ private
    function Image (This : Var_Ref) return String
    is (Image (This.Ptr.all));
 
+   function Image (This : Element_Type) return String
+   is (Element_Type'Image (This));
+
    --------------
    -- Is_Empty --
    --------------
```

**Comment by mosteo** [comment]
Hi, you're right that the example is not quite correct for Ada pre-2022. To properly instantiate for strings you need something like:
```
   function Image (S : String) return String is (S);
   
   package Opt_Strings is new Optional.Values (String, Image);
```

With Ada2022 enabled, you could probably use `String'Image` directly as the second argument to the package instantiation.

**Comment by mosteo** [comment]
Fixed in 3b6d741c320608579419db9871fe3fd02356d9b6
